byterun/alloc.c

```diff
@@ -134,6 +134,23 @@ CAMLexport value caml_alloc_array(value (*funct)(char const *),
   }
 }
 
+/* [len] is a number of floats */
+CAMLprim value caml_alloc_float_array(mlsize_t len)
```

Comment:
This causes the function to be added to the primitives table in the bytecode runtime, which I think can only be an error here? This should have been `CAMLexport` rather than `CAMLprim`, I think?

Reply:
@dra27 is correct, this should be `CAMLexport`, in particular because it cannot be called from OCaml bytecode (first argument is an unboxed integer).

Reply:
Thanks! It'll be fixed as part of a horrific, I mean glorious, update to the caml headers coming soon to a pull request near you...